src/macos.rs

```diff
@@ -0,0 +1,65 @@
+use std::ptr::addr_of_mut;
+
+use libc::c_int;
+
+use crate::ffi;
+use crate::{HidApi, HidDevice, HidResult};
+
+impl HidApi {
+    /// **`Only available on MacOS`**
+    ///
+    /// Changes the behavior of all further calls that open a new [`HidDevice`]
+    /// like [`HidApi::open`] or [`HidApi::open_path`]. By default on Darwin
+    /// platform all devices opened by [`HidApi`] are opened in exclusive mode.
+    ///
+    /// When `exclusive` is set to:
+    ///   * `false` - all further devices will be opened in non-exclusive mode.
+    ///   * `true` all further devices will be opened in exclusive mode.
+    pub fn set_open_exclusive(&self, exclusive: bool) {
+        unsafe { ffi::macos::hid_darwin_set_open_exclusive(exclusive as c_int) }
+    }
+
+    /// **`Only available on MacOS`**
+    ///
+    /// Get the current opening behavior set by [`HidApi::set_open_exclusive`].
+    pub fn get_open_exclusive(&self) -> bool {
+        unsafe { ffi::macos::hid_darwin_get_open_exclusive() != 0 }
+    }
+}
+
+impl HidDevice {
+    /// **`Only available on MacOS`**
+    ///
+    /// Get the location ID for a [`HidDevice`] device.
+    pub fn get_location_id(&self) -> HidResult<u32> {
+        let mut location_id: u32 = 0;
+
+        let res =
+            unsafe { ffi::macos::hid_darwin_get_location_id(self._hid_device, addr_of_mut!(location_id)) };
```

Comment:
There is no reason to use addr_of_mut!()? Could just obtain a mutable reference to the variable.

Reply:
Done! You can see the [changes here.](https://github.com/ruabmbua/hidapi-rs/compare/02a62044aaa72eeea404f30415fc6f7fd3d50e18..4dac1a20d7b4be3df307040a980a4abdbb21a87b)